src/Bicep.LangServer.UnitTests/Handlers/BicepEditLinterRuleCommandHandlerTests.cs

```diff
@@ -0,0 +1,260 @@
+
+            selectedText.Should().Be("warning", "new rule's level value should be selected when the config file is opened");
+            NormalizeLineEndings(File.ReadAllText(configPath)).Should().Be(expectedConfig);
+        }
+
+        [TestMethod]
+        public async Task IfConfigExists_AndIsInvalid_ThenThrow()
+        {
+            string bicepConfig = @"invalid json";
+
+            var (bicepPath, configPath) = CreateFiles(bicepConfig);
+
+            var server = CreateMockLanguageServer(
+                (ShowDocumentParams @params, CancellationToken token) =>
+                {
+                },
+                new ShowDocumentResult() { Success = true });
+
+            BicepEditLinterRuleCommandHandler bicepEditLinterRuleHandler = new(Serializer, server.Object);
+            await FluentActions
+                .Awaiting(() => bicepEditLinterRuleHandler.Handle(new Uri(bicepPath), "whatever", configPath, CancellationToken.None))
+                .Should()
+                .ThrowAsync<Newtonsoft.Json.JsonException>();
+        }
+
+        [TestMethod]
+        public async Task IfConfigDoesNotExist_ThenCreateAndAddRuleAndSelect()
+        {
+            string expectedConfig = @"{
+  ""analyzers"": {
+    ""core"": {
+      ""rules"": {
+        ""whatever"": {
+          ""level"": ""warning""
+        }
+      }
+    }
+  }
+}";
+
+            var rootFolder = FileHelper.GetUniqueTestOutputPath(TestContext);
+            Directory.CreateDirectory(rootFolder);
+
+            var bicepFolder = Path.Combine(rootFolder, "subfolder");
+            Directory.CreateDirectory(bicepFolder);
+            string bicepPath = Path.Combine(bicepFolder, "main.bicep");
+            File.WriteAllText(bicepPath, "var a = 'hello'");
+
+            string expectedConfigPath = Path.Join(rootFolder, "bicepconfig.json");
+
+            string? selectedText = null;
+            var server = CreateMockLanguageServer(
+                (ShowDocumentParams @params, CancellationToken token) =>
+                {
+                    @params.Uri.GetFileSystemPath().ToLowerInvariant().Should().Be(expectedConfigPath.ToLowerInvariant());
+                    selectedText = GetSelectedTextFromFile(@params.Uri, @params.Selection);
+                },
+                new ShowDocumentResult() { Success = true },
+                new Container<WorkspaceFolder>(new WorkspaceFolder[] { new() { Name = "my workspace", Uri = DocumentUri.File(rootFolder) } }));
+
+            BicepEditLinterRuleCommandHandler bicepEditLinterRuleHandler = new(Serializer, server.Object);
+            await bicepEditLinterRuleHandler.Handle(new Uri(bicepPath), "whatever", "", CancellationToken.None);
+
+            selectedText.Should().Be("warning", "new rule's level value should be selected when the config file is opened");
+            NormalizeLineEndings(File.ReadAllText(expectedConfigPath)).Should().Be(expectedConfig);
```

Comment:
Do we need NormalizeLineEndings? Can we not use BeEquivalentToIgnoringNewlines from https://github.com/Azure/bicep/blob/687319d88207ed4d5e4a47c9a1db4d9dc5c8ec45/src/Bicep.Core.UnitTests/Assertions/StringAssertionsExtensions.cs#L67

Reply:
I was thinking more along the lines of:
https://github.com/Azure/bicep/blob/6afa660cba723c79181b98b80f9965234e94857e/src/Bicep.LangServer.UnitTests/Snippets/SnippetsProviderTests.cs#L175

Reply:
Perfect, thanks.